### ISSUE TITLE
[JENKINS-40439] Move parsers into external library

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/warnings-ng-plugin#2580]
<p>Rather than holding parsers within warnings-plugin, move them to an external library, so that the same parser can be used for multiple plugins.</p>

<p>There is  <a href="https://github.com/tomasbjerre/violations-lib" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib</a> that already duplicates some of the parsers (plus might add a few extra). Would you consider merging the parser logic into that library (and moving it into jenkinsci-organization in Github)?</p>

<p>See <a href="https://github.com/jenkinsci/violations-plugin/issues/88#issuecomment-266990121" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/violations-plugin/issues/88#issuecomment-266990121</a>. I would assume some sort of a merge would be nice here.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tuukkamustonen.jpg" title="tuukkamustonen's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tuukkamustonen">tuukkamustonen</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439" target="_blank">Move parsers into external library</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli">drulli</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: warnings-plugin
<li><b>label(s)</b>: analysis-core-2.0
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-03-10T11:59:07+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 7
<li><b>imported</b>: 2025-12-09
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Rather than holding parsers within warnings-plugin, move them to an external library, so that the same parser can be used for multiple plugins.</p>

<p>There is  <a href="https://github.com/tomasbjerre/violations-lib" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib</a> that already duplicates some of the parsers (plus might add a few extra). Would you consider merging the parser logic into that library (and moving it into jenkinsci-organization in Github)?</p>

<p>See <a href="https://github.com/jenkinsci/violations-plugin/issues/88#issuecomment-266990121" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/violations-plugin/issues/88#issuecomment-266990121</a>. I would assume some sort of a merge would be nice here.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280365&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280365">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That would make sense, yes. Before we can do this step, we need to define a new API for this plugin (or library). </p>

<p>Questions to address:</p>
<ol>
	<li>Does the library depend on Jenkins core? Or is it a plain Java library? Currently the parsers implement ExtensionPoint... Also the serialisation is based on Jenkins XStream. Additionally, the library must be part of a plug-in to get the class loading done.</li>
	<li>How does the object graph of the result look like? I would like to change the current structure of Warning, ParserResult, etc.</li>
	<li>Is there a way to obtain only the number of warnings rather than the actual warnings?</li>
</ol>


<p>PS for a bachelor thesis of one of my students I created a prototype module to see what is actually required: <a href="https://github.com/jenkinsci/analysis-model" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model</a> </p>
</pre>
</details>

</ul>
</details>

<p>That would make sense, yes. Before we can do this step, we need to define a new API for this plugin (or library). </p>

<p>Questions to address:</p>
<ol>
	<li>Does the library depend on Jenkins core? Or is it a plain Java library? Currently the parsers implement ExtensionPoint... Also the serialisation is based on Jenkins XStream. Additionally, the library must be part of a plug-in to get the class loading done.</li>
	<li>How does the object graph of the result look like? I would like to change the current structure of Warning, ParserResult, etc.</li>
	<li>Is there a way to obtain only the number of warnings rather than the actual warnings?</li>
</ol>


<p>PS for a bachelor thesis of one of my students I created a prototype module to see what is actually required: <a href="https://github.com/jenkinsci/analysis-model" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model</a> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tomasbjerre.png" title="tomasbjerre's avatar" /> tomasbjerre:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280382&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280382">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>1. No. Just SLF4J <a href="https://github.com/tomasbjerre/violations-lib/blob/master/build.gradle#L43" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/blob/master/build.gradle#L43</a> .
2. Its a list of Violations <a href="https://github.com/tomasbjerre/violations-lib/blob/master/src/main/java/se/bjurr/violations/lib/model/Violation.java" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/blob/master/src/main/java/se/bjurr/violations/lib/model/Violation.java</a>
3. No.</p>

<p>When it comes to defining an API I am pretty much happy with what I have. But I am open for suggestions!</p>

<p>I took a quick look at analysis-model. The amount of dependencies are a show stopper for me. I would rather spend time on making the parsing algorithms use only functionality from the standard library (1.7).</p>
</pre>
</details>

</ul>
</details>

<p>1. No. Just SLF4J <a href="https://github.com/tomasbjerre/violations-lib/blob/master/build.gradle#L43" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/blob/master/build.gradle#L43</a> .<br/>
2. Its a list of Violations <a href="https://github.com/tomasbjerre/violations-lib/blob/master/src/main/java/se/bjurr/violations/lib/model/Violation.java" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/blob/master/src/main/java/se/bjurr/violations/lib/model/Violation.java</a><br/>
3. No.</p>

<p>When it comes to defining an API I am pretty much happy with what I have. But I am open for suggestions!</p>

<p>I took a quick look at analysis-model. The amount of dependencies are a show stopper for me. I would rather spend time on making the parsing algorithms use only functionality from the standard library (1.7).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280415&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280415">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>1a. No Jenkins dependencies means a lot of duplication: a wrapper for each violation model class in analysis-core (otherwise serialization/classloading will not work: maybe this works when exporting the violation-lib classes in analysis-core). Reimplementation of extension point pattern (using a registry), etc. It is possible going this way but this will require a lot of extra work (and testing). The question is what you loose if we have a dependency to Jenkins. 
1b. No other dependencies means a lot of rework: several parsers require different libraries. What is the reason behind limiting the dependencies to just logging (why is this a show stopper)? Why Java 7 and not 8?
2. Currently this is almost the same in analysis-core (plus a lot of additional utility classes). This is one thing I would change if I could, this model is quite limited when used in conjunction with DB backend, UI filtering and paging.
3. This is something requested by some users, would be nice if it could be integrated somehow if I totally change the underlying model</p>
</pre>
</details>

</ul>
</details>

<p>1a. No Jenkins dependencies means a lot of duplication: a wrapper for each violation model class in analysis-core (otherwise serialization/classloading will not work: maybe this works when exporting the violation-lib classes in analysis-core). Reimplementation of extension point pattern (using a registry), etc. It is possible going this way but this will require a lot of extra work (and testing). The question is what you loose if we have a dependency to Jenkins. <br/>
1b. No other dependencies means a lot of rework: several parsers require different libraries. What is the reason behind limiting the dependencies to just logging (why is this a show stopper)? Why Java 7 and not 8?<br/>
2. Currently this is almost the same in analysis-core (plus a lot of additional utility classes). This is one thing I would change if I could, this model is quite limited when used in conjunction with DB backend, UI filtering and paging.<br/>
3. This is something requested by some users, would be nice if it could be integrated somehow if I totally change the underlying model</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tomasbjerre.png" title="tomasbjerre's avatar" /> tomasbjerre:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280447&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280447">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>1a I use the library also in Gradle and Maven plugins. Others may use it in whatever application they want, perhaps another CI server. Having a dependency to Jenins just does not make sense for a library whose only task is to read and parse text-files.
1b 
 I prefer to make it Java 7 compatible because its still used by alot of people.
 Unnecessary libraries on the classpath will be a problem for users of the library. For example, they may already have another version of Apache commons already on the classpath. It may be there because of another dependency or because they need it themselves. They would have to exclude dependencies and sometimes it will be impossible to resolve the situation and find a version that is compatible with all libraries using it.
 I have been able to implement alot of parsers this way, the library is very clean, very easy to work with and understand for anyone familiar with Java. I'd like to keep it that way.</p>

<p>2 That to me sounds like something other then reading and parsing text files. Not a problem that should be solved by the library.</p>
</pre>
</details>

</ul>
</details>

<p>1a I use the library also in Gradle and Maven plugins. Others may use it in whatever application they want, perhaps another CI server. Having a dependency to Jenins just does not make sense for a library whose only task is to read and parse text-files.<br/>
1b <br/>
 I prefer to make it Java 7 compatible because its still used by alot of people.<br/>
 Unnecessary libraries on the classpath will be a problem for users of the library. For example, they may already have another version of Apache commons already on the classpath. It may be there because of another dependency or because they need it themselves. They would have to exclude dependencies and sometimes it will be impossible to resolve the situation and find a version that is compatible with all libraries using it.<br/>
 I have been able to implement alot of parsers this way, the library is very clean, very easy to work with and understand for anyone familiar with Java. I'd like to keep it that way.</p>

<p>2 That to me sounds like something other then reading and parsing text files. Not a problem that should be solved by the library.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tuukkamustonen.jpg" title="tuukkamustonen's avatar" /> tuukkamustonen:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280534&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280534">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm not author of anything here, but here's my two cents:</p>

<p>1. I interpret Tomas' statement so that he's not exactly against dependency libraries (that would be weird, as <tt>violations-lib</tt> is a dependency for other projects).</p>

<p>The problem of conflicting requirements versions between multiple libraries is common. I not user of NodeJS, but as I've understood it solves that by having isolated requirements for each library (and also means lots of "wasted" disk space and potentially higher memory consumption... just assuming). In other languages you typically specify minimum version instead of pinning to an exact one. Of course, newer versions <b>may</b> break compatibility, so it's not fool proof, but you can constraint maximum version also. Would that work as a compromise, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tomasbjerre" class="user-hover" rel="tomasbjerre">tomasbjerre</a>?</p>

<p>I do agree that this library should not depend on jenkins libraries (<tt>analysis-core, violations, script-security</tt>). In addition to them, there's only <tt>commons-digester3</tt> and I don't see it bad (commons and guava libraries are useful, if they cut the amount of code needed, just use them, imo). Other dependencies are test-scope only, and would probably mostly vanish if you get rid of the jenkins-related libraries.</p>

<blockquote>
<p>3. Is there a way to obtain only the number of warnings rather than the actual warnings?</p></blockquote>

<p>Can't you just <tt>.size()</tt> the results? You have to run the full analysis anyway. Maybe I'm missing the point here...</p>
</pre>
</details>

</ul>
</details>

<p>I'm not author of anything here, but here's my two cents:</p>

<p>1. I interpret Tomas' statement so that he's not exactly against dependency libraries (that would be weird, as <tt>violations-lib</tt> is a dependency for other projects).</p>

<p>The problem of conflicting requirements versions between multiple libraries is common. I not user of NodeJS, but as I've understood it solves that by having isolated requirements for each library (and also means lots of "wasted" disk space and potentially higher memory consumption... just assuming). In other languages you typically specify minimum version instead of pinning to an exact one. Of course, newer versions <b>may</b> break compatibility, so it's not fool proof, but you can constraint maximum version also. Would that work as a compromise, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tomasbjerre" class="user-hover" rel="tomasbjerre">tomasbjerre</a>?</p>

<p>I do agree that this library should not depend on jenkins libraries (<tt>analysis-core, violations, script-security</tt>). In addition to them, there's only <tt>commons-digester3</tt> and I don't see it bad (commons and guava libraries are useful, if they cut the amount of code needed, just use them, imo). Other dependencies are test-scope only, and would probably mostly vanish if you get rid of the jenkins-related libraries.</p>

<blockquote>
<p>3. Is there a way to obtain only the number of warnings rather than the actual warnings?</p></blockquote>

<p>Can't you just <tt>.size()</tt> the results? You have to run the full analysis anyway. Maybe I'm missing the point here...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tomasbjerre.png" title="tomasbjerre's avatar" /> tomasbjerre:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280537&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280537">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>1. Sort of yes.</p>

<p>If you take Guava for example. I have been using this method alot:
<a href="https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Objects.html#firstNonNull(T" class="external-link" target="_blank" rel="nofollow noopener">https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Objects.html#firstNonNull(T</a>, T)
As you can read there it says "scheduled for removal in June 2016". If the library was using that method then anyone trying to use the library in an application with latest Guava would get ClassNotFoundException if they have latest Guava on classpath, or they would have to downgrade their Guava to the older one used by the library.</p>

<p>Also commons-digester3 (<a href="https://repo1.maven.org/maven2/org/apache/commons/commons-digester3/3.2/commons-digester3-3.2.pom" class="external-link" target="_blank" rel="nofollow noopener">https://repo1.maven.org/maven2/org/apache/commons/commons-digester3/3.2/commons-digester3-3.2.pom</a>) adds cglib, commons-beanutils, commons-logging + whatever libraries they depend on.</p>

<p>A good library does not force the user of the library into using a specific version of a utility library like this. I dont see the need to use anything else but the standard library here. Show me an example and I may change my mind =)</p>

<p>3. Yes, .size() would work.</p>
</pre>
</details>

</ul>
</details>

<p>1. Sort of yes.</p>

<p>If you take Guava for example. I have been using this method alot:<br/>
<a href="https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Objects.html#firstNonNull(T" class="external-link" target="_blank" rel="nofollow noopener">https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Objects.html#firstNonNull(T</a>, T)<br/>
As you can read there it says "scheduled for removal in June 2016". If the library was using that method then anyone trying to use the library in an application with latest Guava would get ClassNotFoundException if they have latest Guava on classpath, or they would have to downgrade their Guava to the older one used by the library.</p>

<p>Also commons-digester3 (<a href="https://repo1.maven.org/maven2/org/apache/commons/commons-digester3/3.2/commons-digester3-3.2.pom" class="external-link" target="_blank" rel="nofollow noopener">https://repo1.maven.org/maven2/org/apache/commons/commons-digester3/3.2/commons-digester3-3.2.pom</a>) adds cglib, commons-beanutils, commons-logging + whatever libraries they depend on.</p>

<p>A good library does not force the user of the library into using a specific version of a utility library like this. I dont see the need to use anything else but the standard library here. Show me an example and I may change my mind =)</p>

<p>3. Yes, .size() would work.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280719&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280719">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I think it is possible to remove the Jenkins dependencies, it just will require some additional work: currently the model is part of analysis-core, the parsers are in the warnings plug-in. If we move the parsers to a separate library then the model needs to be duplicated (and the extension point needs to be replaced by a registry) and after the parsing the model needs to be transformed to the Jenkins model.</p>

<p>On the other hand I'm not so dogmatic on using other libraries. (BTW: if you don't like libraries why don't you use java logging and remove slf4j, too?). I think if I can use a library that reduces the amount of code (and effort) I'll use it. Also I don't care much about things that might happen in future - if it really happens that we have a library dependency problem then we still have the possibility to change it (and <a href="https://issues.jenkins.io/browse/JENKINS-40439" title="Move parsers into external library" class="issue-link" data-issue-key="JENKINS-40439"><del>JENKINS-40439</del></a> is exactly such a change request, isn't it?). Moreover, there are some ways to hide dependencies - if required, e.g. I'm using the maven shade plug-in to hide some of the dependencies in my findbugs parser wrapper that are in conflict with Jenkins dependencies. </p>

<p>Several parsers now use digester and I see no <b>real</b> requirement to reimplement all these parsers just to drop the dependency (of course these parsers could use Java 7 classes only, but why should be rewrite working code?). And this would be even more problematic for the FindBugs parser: here the FindBugs team already provides one (which in contrast to your parser also handles some edge cases), why should I implement another one? I would like to spend this effort somewhere else...</p>

<p>(Just a side note to 3.: the reason for this requirement is from some users having files with thousands of warnings. Parsing these files into a List of warnings in memory is quite slow so using size() on this list will not help. But this is actually not a high priority requirement.)</p>
</pre>
</details>

</ul>
</details>

<p>I think it is possible to remove the Jenkins dependencies, it just will require some additional work: currently the model is part of analysis-core, the parsers are in the warnings plug-in. If we move the parsers to a separate library then the model needs to be duplicated (and the extension point needs to be replaced by a registry) and after the parsing the model needs to be transformed to the Jenkins model.</p>

<p>On the other hand I'm not so dogmatic on using other libraries. (BTW: if you don't like libraries why don't you use java logging and remove slf4j, too?). I think if I can use a library that reduces the amount of code (and effort) I'll use it. Also I don't care much about things that might happen in future - if it really happens that we have a library dependency problem then we still have the possibility to change it (and <a href="https://issue-redirect.jenkins.io/issue/40439" title="Move parsers into external library" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/40439">JENKINS-40439</a>"><del>JENKINS-40439</del></a> is exactly such a change request, isn't it?). Moreover, there are some ways to hide dependencies - if required, e.g. I'm using the maven shade plug-in to hide some of the dependencies in my findbugs parser wrapper that are in conflict with Jenkins dependencies. </p>

<p>Several parsers now use digester and I see no <b>real</b> requirement to reimplement all these parsers just to drop the dependency (of course these parsers could use Java 7 classes only, but why should be rewrite working code?). And this would be even more problematic for the FindBugs parser: here the FindBugs team already provides one (which in contrast to your parser also handles some edge cases), why should I implement another one? I would like to spend this effort somewhere else...</p>

<p>(Just a side note to 3.: the reason for this requirement is from some users having files with thousands of warnings. Parsing these files into a List of warnings in memory is quite slow so using size() on this list will not help. But this is actually not a high priority requirement.)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tomasbjerre.png" title="tomasbjerre's avatar" /> tomasbjerre:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280728&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280728">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>"if you don't like libraries"</p>

<p>I do like libraries. I dont like libraries that polutes my classpath. I dont want to force users of my library into using a specific version of some other library.</p>

<p>"why don't you use java logging and remove slf4j, too?"</p>

<p>I was using slf4j-api, and only the basic functionality. That is very unlikely to casue problems. But on the other hand I only do very little logging so I actually removed slf4j-api too.</p>

<p>The FindBugs report is tricky, yes. If there is a findbugs-parser ready and working using digester then that, to me, is a valid argument for having digester on the classpath. And use shading to avoid classpath collisions.</p>
</pre>
</details>

</ul>
</details>

<p>"if you don't like libraries"</p>

<p>I do like libraries. I dont like libraries that polutes my classpath. I dont want to force users of my library into using a specific version of some other library.</p>

<p>"why don't you use java logging and remove slf4j, too?"</p>

<p>I was using slf4j-api, and only the basic functionality. That is very unlikely to casue problems. But on the other hand I only do very little logging so I actually removed slf4j-api too.</p>

<p>The FindBugs report is tricky, yes. If there is a findbugs-parser ready and working using digester then that, to me, is a valid argument for having digester on the classpath. And use shading to avoid classpath collisions.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280849&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280849">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>BTW: who will depend on this library? Is that already clear? It would help to see their requirements... Currently we have the warnings-plugin and the violation-comments-to-github-plugin in Jenkins.</p>
</pre>
</details>

</ul>
</details>

<p>BTW: who will depend on this library? Is that already clear? It would help to see their requirements... Currently we have the warnings-plugin and the violation-comments-to-github-plugin in Jenkins.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tomasbjerre.png" title="tomasbjerre's avatar" /> tomasbjerre:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280857&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280857">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://github.com/jenkinsci/violation-comments-to-stash-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/violation-comments-to-stash-plugin</a>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin</a>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin</a></p>

<p>But I think I've made my points here. Don't have anything more to add to this discussion.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tuukkamustonen" class="user-hover" rel="tuukkamustonen">tuukkamustonen</a>: We clearly have very different views on how this should be solved so I don't see this happening. I think what I have in violations-lib is the perfect approach and if you need any parser implemented, just open an issue there! =)</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://github.com/jenkinsci/violation-comments-to-stash-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/violation-comments-to-stash-plugin</a><br/>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin</a><br/>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin</a></p>

<p>But I think I've made my points here. Don't have anything more to add to this discussion.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tuukkamustonen" class="user-hover" rel="tuukkamustonen">tuukkamustonen</a>: We clearly have very different views on how this should be solved so I don't see this happening. I think what I have in violations-lib is the perfect approach and if you need any parser implemented, just open an issue there! =)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tuukkamustonen.jpg" title="tuukkamustonen's avatar" /> tuukkamustonen:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280864&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280864">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Don't give up just yet - I think it the discussion has been positive and constructive. I feel you both possess great open-minded attitude, it's just natural that you have different matter of taste, but we're getting close!</p>

<p>About duplication:</p>

<blockquote>
<p>If we move the parsers to a separate library then the model needs to be duplicated (and the extension point needs to be replaced by a registry) and after the parsing the model needs to be transformed to the Jenkins model.</p></blockquote>

<p>I don't know the technical details, but I think it's just natural that you need some sort of an adapter that maps each parser into the Jenkins plugin. Yeah, it will add some code (but you shouldn't need to duplicate code). That's the price you must pay for separation of concerns. However, you will collect the benefits later on.</p>

<p>Would it be possible to somehow, dynamically resolve and construct these adapters, so that you wouldn't have to explicitly write each?</p>

<p>About the dependency libraries:</p>

<blockquote>
<p>The FindBugs report is tricky, yes. If there is a findbugs-parser ready and working using digester then that, to me, is a valid argument for having digester on the classpath. And use shading to avoid classpath collisions.</p></blockquote>

<p>Sounds good. I'm sure you'll come to terms about dependency libraries if you work on this: use what you need, omit what you don't. You'll start with what you got (minus the Jenkins dependencies), and if Tomas wants less dependencies, I'm sure he's quick to refactor the code.</p>

<p>Besides, you can still depend on more dependencies in warnings-plugin. It's just violations-lib that should be minimal.</p>

<blockquote>
<p>(of course these parsers could use Java 7 classes only, but why should be rewrite working code?</p></blockquote>

<p>Same here, I believe Tomas is a pretty productive guy, so he'll quickly do this <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> Simplifying and working on technical debt every now and then makes sense (although, I don't think this categorizes as "technical debt", but anyway).</p>

<p>If you got tests there, it's less likely that things will break.</p>

<blockquote>
<p>(Just a side note to 3.: the reason for this requirement is from some users having files with thousands of warnings. Parsing these files into a List of warnings in memory is quite slow so using size() on this list will not help. But this is actually not a high priority requirement.)</p></blockquote>

<p>Tell them they should really fix their code <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>I know I'm over-simplifying things but you should "just":</p>

<p>1. Tomas copies/merges the parsers from warnings-plugin into violations-lib.
2. Agree/ensure on the parser API in violations-lib.
3. Ulli switches from built-in parsers into violations-lib. Put whatever you need there, we'll have just single parser implementations in the end, which is the goal of this ticket. You can tune things later on, if need be.</p>

<p>Even if steps 2-3 never succeed, there's benefit in step one, as violations-lib will get quite a few parsers that are now missing.</p>
</pre>
</details>

</ul>
</details>

<p>Don't give up just yet - I think it the discussion has been positive and constructive. I feel you both possess great open-minded attitude, it's just natural that you have different matter of taste, but we're getting close!</p>

<p>About duplication:</p>

<blockquote>
<p>If we move the parsers to a separate library then the model needs to be duplicated (and the extension point needs to be replaced by a registry) and after the parsing the model needs to be transformed to the Jenkins model.</p></blockquote>

<p>I don't know the technical details, but I think it's just natural that you need some sort of an adapter that maps each parser into the Jenkins plugin. Yeah, it will add some code (but you shouldn't need to duplicate code). That's the price you must pay for separation of concerns. However, you will collect the benefits later on.</p>

<p>Would it be possible to somehow, dynamically resolve and construct these adapters, so that you wouldn't have to explicitly write each?</p>

<p>About the dependency libraries:</p>

<blockquote>
<p>The FindBugs report is tricky, yes. If there is a findbugs-parser ready and working using digester then that, to me, is a valid argument for having digester on the classpath. And use shading to avoid classpath collisions.</p></blockquote>

<p>Sounds good. I'm sure you'll come to terms about dependency libraries if you work on this: use what you need, omit what you don't. You'll start with what you got (minus the Jenkins dependencies), and if Tomas wants less dependencies, I'm sure he's quick to refactor the code.</p>

<p>Besides, you can still depend on more dependencies in warnings-plugin. It's just violations-lib that should be minimal.</p>

<blockquote>
<p>(of course these parsers could use Java 7 classes only, but why should be rewrite working code?</p></blockquote>

<p>Same here, I believe Tomas is a pretty productive guy, so he'll quickly do this <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> Simplifying and working on technical debt every now and then makes sense (although, I don't think this categorizes as "technical debt", but anyway).</p>

<p>If you got tests there, it's less likely that things will break.</p>

<blockquote>
<p>(Just a side note to 3.: the reason for this requirement is from some users having files with thousands of warnings. Parsing these files into a List of warnings in memory is quite slow so using size() on this list will not help. But this is actually not a high priority requirement.)</p></blockquote>

<p>Tell them they should really fix their code <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>I know I'm over-simplifying things but you should "just":</p>

<p>1. Tomas copies/merges the parsers from warnings-plugin into violations-lib.<br/>
2. Agree/ensure on the parser API in violations-lib.<br/>
3. Ulli switches from built-in parsers into violations-lib. Put whatever you need there, we'll have just single parser implementations in the end, which is the goal of this ticket. You can tune things later on, if need be.</p>

<p>Even if steps 2-3 never succeed, there's benefit in step one, as violations-lib will get quite a few parsers that are now missing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280924&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280924">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote>
<p><a href="https://github.com/jenkinsci/violation-comments-to-stash-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/violation-comments-to-stash-plugin</a>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin</a>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin</a></p></blockquote>
<p>But these are only projects under your control, here <b>you</b> can manage the library versions. You did mention 'users of the project' in <a href="https://issues.jenkins-ci.org/browse/JENKINS-40439?focusedCommentId=280447&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-280447" class="external-link" target="_blank" rel="nofollow noopener">a previous comment</a>, are these potential users or are there already other people using the library?  </p>
<blockquote>
<p>But I think I've made my points here. Don't have anything more to add to this discussion. We clearly have very different views on how this should be solved so I don't see this happening. </p></blockquote>
<p>Well, if you are not open to a discussion, than it will be so. When I spend my spare time on a project I want to make sure that it is a good investment that helps to fulfill some <b>real</b> requirements. I won't just change a lot of code just because the design is better or cleaner. This violates the YAGNI and KISS principles I'm following. </p>

<p>Actually a design or architecture is a solution for a given set of requirements. The only <b>real</b> requirement I'm seeing, is to provide a library of parsers that has no Jenkins dependencies. This is something we can work on. </p>
</pre>
</details>

</ul>
</details>

<blockquote>
<p><a href="https://github.com/jenkinsci/violation-comments-to-stash-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/violation-comments-to-stash-plugin</a><br/>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-maven-plugin</a><br/>
<a href="https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violation-comments-to-github-gradle-plugin</a></p></blockquote>
<p>But these are only projects under your control, here <b>you</b> can manage the library versions. You did mention 'users of the project' in <a href="https://issues.jenkins-ci.org/browse/JENKINS-40439?focusedCommentId=280447&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-280447" class="external-link" target="_blank" rel="nofollow noopener">a previous comment</a>, are these potential users or are there already other people using the library?  </p>
<blockquote>
<p>But I think I've made my points here. Don't have anything more to add to this discussion. We clearly have very different views on how this should be solved so I don't see this happening. </p></blockquote>
<p>Well, if you are not open to a discussion, than it will be so. When I spend my spare time on a project I want to make sure that it is a good investment that helps to fulfill some <b>real</b> requirements. I won't just change a lot of code just because the design is better or cleaner. This violates the YAGNI and KISS principles I'm following. </p>

<p>Actually a design or architecture is a solution for a given set of requirements. The only <b>real</b> requirement I'm seeing, is to provide a library of parsers that has no Jenkins dependencies. This is something we can work on. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=280927&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-280927">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote>
<p>I don't know the technical details, but I think it's just natural that you need some sort of an adapter that maps each parser into the Jenkins plugin. Yeah, it will add some code (but you shouldn't need to duplicate code). That's the price you must pay for separation of concerns. However, you will collect the benefits later on.</p></blockquote>
<p>Yes, it's no duplication, just a wrapper for each parser and model class. I've done this for the parsers of the violations plugin already quite some time ago, see <a href="https://github.com/jenkinsci/warnings-plugin/blob/master/src/main/java/hudson/plugins/warnings/parser/ViolationsAdapter.java" class="external-link" target="_blank" rel="nofollow noopener">ViolationsAdapter</a>. (Note that this implementation is working only partly, since the parsers of the violations plug-in are not serializable and will not work on slave builds.)  </p>
<blockquote>
<p>Would it be possible to somehow, dynamically resolve and construct these adapters, so that you wouldn't have to explicitly write each?</p></blockquote>
<p>Yes, see <a href="https://github.com/jenkinsci/warnings-plugin/blob/master/src/main/java/hudson/plugins/warnings/parser/ViolationsRegistry.java" class="external-link" target="_blank" rel="nofollow noopener">ViolationsRegistry</a>.</p>
<blockquote>
<p>Sounds good. I'm sure you'll come to terms about dependency libraries if you work on this: use what you need, omit what you don't. You'll start with what you got (minus the Jenkins dependencies), and if Tomas wants less dependencies, I'm sure he's quick to refactor the code.
Besides, you can still depend on more dependencies in warnings-plugin. It's just violations-lib that should be minimal.
If you got tests there, it's less likely that things will break.</p></blockquote>
<p>Yes, if you want to replace the parsers with something simpler, go ahead<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> I just don't have the time to replace existing and working (and tested) code with something that produces the same results in a different way <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>
<blockquote>
<p>Tell them they should really fix their code </p></blockquote>
<p>I tried but this issue pops up from time to time <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<blockquote>
<p>I know I'm over-simplifying things but you should "just":
1. Tomas copies/merges the parsers from warnings-plugin into violations-lib.
2. Agree/ensure on the parser API in violations-lib.
3. Ulli switches from built-in parsers into violations-lib. Put whatever you need there, we'll have just single parser implementations in the end, which is the goal of this ticket. You can tune things later on, if need be.
Even if steps 2-3 never succeed, there's benefit in step one, as violations-lib will get quite a few parsers that are now missing.</p></blockquote>
<p>I think this is a feasible approach. I think 2 should be done before 1: otherwise a lot of things will break afterwards.</p>
</pre>
</details>

</ul>
</details>

<blockquote>
<p>I don't know the technical details, but I think it's just natural that you need some sort of an adapter that maps each parser into the Jenkins plugin. Yeah, it will add some code (but you shouldn't need to duplicate code). That's the price you must pay for separation of concerns. However, you will collect the benefits later on.</p></blockquote>
<p>Yes, it's no duplication, just a wrapper for each parser and model class. I've done this for the parsers of the violations plugin already quite some time ago, see <a href="https://github.com/jenkinsci/warnings-plugin/blob/master/src/main/java/hudson/plugins/warnings/parser/ViolationsAdapter.java" class="external-link" target="_blank" rel="nofollow noopener">ViolationsAdapter</a>. (Note that this implementation is working only partly, since the parsers of the violations plug-in are not serializable and will not work on slave builds.)  </p>
<blockquote>
<p>Would it be possible to somehow, dynamically resolve and construct these adapters, so that you wouldn't have to explicitly write each?</p></blockquote>
<p>Yes, see <a href="https://github.com/jenkinsci/warnings-plugin/blob/master/src/main/java/hudson/plugins/warnings/parser/ViolationsRegistry.java" class="external-link" target="_blank" rel="nofollow noopener">ViolationsRegistry</a>.</p>
<blockquote>
<p>Sounds good. I'm sure you'll come to terms about dependency libraries if you work on this: use what you need, omit what you don't. You'll start with what you got (minus the Jenkins dependencies), and if Tomas wants less dependencies, I'm sure he's quick to refactor the code.<br/>
Besides, you can still depend on more dependencies in warnings-plugin. It's just violations-lib that should be minimal.<br/>
If you got tests there, it's less likely that things will break.</p></blockquote>
<p>Yes, if you want to replace the parsers with something simpler, go ahead<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> I just don't have the time to replace existing and working (and tested) code with something that produces the same results in a different way <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>
<blockquote>
<p>Tell them they should really fix their code </p></blockquote>
<p>I tried but this issue pops up from time to time <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<blockquote>
<p>I know I'm over-simplifying things but you should "just":<br/>
1. Tomas copies/merges the parsers from warnings-plugin into violations-lib.<br/>
2. Agree/ensure on the parser API in violations-lib.<br/>
3. Ulli switches from built-in parsers into violations-lib. Put whatever you need there, we'll have just single parser implementations in the end, which is the goal of this ticket. You can tune things later on, if need be.<br/>
Even if steps 2-3 never succeed, there's benefit in step one, as violations-lib will get quite a few parsers that are now missing.</p></blockquote>
<p>I think this is a feasible approach. I think 2 should be done before 1: otherwise a lot of things will break afterwards.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tuukkamustonen.jpg" title="tuukkamustonen's avatar" /> tuukkamustonen:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=286376&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-286376">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Well... unfortunately I think this didn't fly. Should probably be closed?</p>
</pre>
</details>

</ul>
</details>

<p>Well... unfortunately I think this didn't fly. Should probably be closed?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=286753&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-286753">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Please let it open. Maybe I can at least integrate some of the parsers into the warnings plug-in in the same style I am using the parsers of the violations plug-in. </p>

<p>Which parsers are actually missing that should be recognized by Jenkins?</p>
</pre>
</details>

</ul>
</details>

<p>Please let it open. Maybe I can at least integrate some of the parsers into the warnings plug-in in the same style I am using the parsers of the violations plug-in. </p>

<p>Which parsers are actually missing that should be recognized by Jenkins?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tuukkamustonen.jpg" title="tuukkamustonen's avatar" /> tuukkamustonen:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=286759&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-286759">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Well, Tomas just added mypy (<a href="https://github.com/tomasbjerre/violations-lib/issues/13" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/issues/13</a>) and pydocstyle (<a href="https://github.com/tomasbjerre/violations-lib/issues/12" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/issues/12</a>) that are not available in warnings.</p>

<p>I have been using custom parsers (via Manage jenkins &gt; Configure system) for those with warnings plugin so I haven't ended up submitting anything... would be nice to have them directly in warnings plugin, of course.</p>

<p>I am personally not missing anything else <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Well, Tomas just added mypy (<a href="https://github.com/tomasbjerre/violations-lib/issues/13" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/issues/13</a>) and pydocstyle (<a href="https://github.com/tomasbjerre/violations-lib/issues/12" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/tomasbjerre/violations-lib/issues/12</a>) that are not available in warnings.</p>

<p>I have been using custom parsers (via Manage jenkins > Configure system) for those with warnings plugin so I haven't ended up submitting anything... would be nice to have them directly in warnings plugin, of course.</p>

<p>I am personally not missing anything else <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pedrofurlanetto:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=289012&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-289012">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is there any update about CodeNarc parser?</p>
</pre>
</details>

</ul>
</details>

<p>Is there any update about CodeNarc parser?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tuukkamustonen.jpg" title="tuukkamustonen's avatar" /> tuukkamustonen:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=318222&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-318222">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hmm, this just moved to In Progress. Is there something going on?</p>
</pre>
</details>

</ul>
</details>

<p>Hmm, this just moved to In Progress. Is there something going on?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=318260&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-318260">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, this is the reason behind the state change <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>It is part of the analysis-core refactoring, see <a href="https://wiki.jenkins.io/display/JENKINS/Static+Analysis+in+Pipelines" class="external-link" target="_blank" rel="nofollow noopener">wiki</a> for details.</p>
</pre>
</details>

</ul>
</details>

<p>Yes, this is the reason behind the state change <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>It is part of the analysis-core refactoring, see <a href="https://wiki.jenkins.io/display/JENKINS/Static+Analysis+in+Pipelines" class="external-link" target="_blank" rel="nofollow noopener">wiki</a> for details.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=330830&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330830">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Ulli Hafner
Path:
 pom.xml
 src/main/java/edu/hm/hafner/analysis/parser/ResharperInspectCodeParser.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/AbstractViolationAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/AndroidLintParserAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/CodeNarcAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/CppCheckAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/DocFxAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/ErrorProneAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/Flake8Adapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/JsHintAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/KlocWorkAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/MyPyAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/PitAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/PyDocStyleAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/ResharperInspectCodeAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/XmlLintAdapter.java
 src/main/java/edu/hm/hafner/analysis/parser/violations/ZptLintAdapter.java
 src/test/java/edu/hm/hafner/analysis/parser/ResharperInspectCodeParserTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/AndroidLintParserAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/CodeNarcAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/CppCheckAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/DocFxAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/ErrorProneAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/Flake8AdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/JsHintAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/KlocWorkAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/MyPyAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/PitAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/PyDocStyleAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/ResharperInspectCodeAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/XmlLintAdapterTest.java
 src/test/java/edu/hm/hafner/analysis/parser/violations/ZptLintAdapterTest.java
 src/test/resources/edu/hm/hafner/analysis/parser/ResharperInspectCode.xml
 src/test/resources/edu/hm/hafner/analysis/parser/codenarc/CodeNarcReport2.xml
 src/test/resources/edu/hm/hafner/analysis/parser/codenarc/CodeNarcXmlReport.xml
 src/test/resources/edu/hm/hafner/analysis/parser/findbugs/spotbugsXml.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/ResharperInspectCode.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/android-lint.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/codeNarc.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/cppcheck.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/docfx.json
 src/test/resources/edu/hm/hafner/analysis/parser/violations/error-prone.log
 src/test/resources/edu/hm/hafner/analysis/parser/violations/flake8.txt
 src/test/resources/edu/hm/hafner/analysis/parser/violations/jshint.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/klocwork.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/mypy.txt
 src/test/resources/edu/hm/hafner/analysis/parser/violations/pit.xml
 src/test/resources/edu/hm/hafner/analysis/parser/violations/pydocstyle.txt
 src/test/resources/edu/hm/hafner/analysis/parser/violations/xmllint.txt
 src/test/resources/edu/hm/hafner/analysis/parser/violations/zptlint.log
<a href="http://jenkins-ci.org/commit/analysis-model/f22e29dd33c93cdae9fd228bbce7bff9a2eac517" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/analysis-model/f22e29dd33c93cdae9fd228bbce7bff9a2eac517</a>
Log:
  Let all parsers not depend on Jenkins anymore.</p>

<p>Integrated all parsers into new module analysis-model. Added adapter
classes for parsers of violations-lib.</p>
<ul class="alternate" type="square">
	<li><span class="error">&#91;FIXED JENKINS-17434&#93;</span></li>
	<li><span class="error">&#91;FIXED JENKINS-40439&#93;</span></li>
</ul>



</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Ulli Hafner<br/>
Path:<br/>
 pom.xml<br/>
 src/main/java/edu/hm/hafner/analysis/parser/ResharperInspectCodeParser.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/AbstractViolationAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/AndroidLintParserAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/CodeNarcAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/CppCheckAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/DocFxAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/ErrorProneAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/Flake8Adapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/JsHintAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/KlocWorkAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/MyPyAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/PitAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/PyDocStyleAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/ResharperInspectCodeAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/XmlLintAdapter.java<br/>
 src/main/java/edu/hm/hafner/analysis/parser/violations/ZptLintAdapter.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/ResharperInspectCodeParserTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/AndroidLintParserAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/CodeNarcAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/CppCheckAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/DocFxAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/ErrorProneAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/Flake8AdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/JsHintAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/KlocWorkAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/MyPyAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/PitAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/PyDocStyleAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/ResharperInspectCodeAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/XmlLintAdapterTest.java<br/>
 src/test/java/edu/hm/hafner/analysis/parser/violations/ZptLintAdapterTest.java<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/ResharperInspectCode.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/codenarc/CodeNarcReport2.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/codenarc/CodeNarcXmlReport.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/findbugs/spotbugsXml.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/ResharperInspectCode.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/android-lint.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/codeNarc.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/cppcheck.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/docfx.json<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/error-prone.log<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/flake8.txt<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/jshint.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/klocwork.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/mypy.txt<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/pit.xml<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/pydocstyle.txt<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/xmllint.txt<br/>
 src/test/resources/edu/hm/hafner/analysis/parser/violations/zptlint.log<br/>
<a href="http://jenkins-ci.org/commit/analysis-model/f22e29dd33c93cdae9fd228bbce7bff9a2eac517" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/analysis-model/f22e29dd33c93cdae9fd228bbce7bff9a2eac517</a><br/>
Log:<br/>
  Let all parsers not depend on Jenkins anymore.</p>

<p>Integrated all parsers into new module analysis-model. Added adapter<br/>
classes for parsers of violations-lib.</p>
<ul class="alternate" type="square">
	<li><span class="error">&#91;FIXED <a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/17434">JENKINS-17434</a>&#93;</span></li>
	<li><span class="error">&#91;FIXED <a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/40439">JENKINS-40439</a>&#93;</span></li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=330832&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330832">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Ulli Hafner
Path:
 src/main/java/io/jenkins/plugins/analysis/warnings/AndroidLint.java
 src/main/java/io/jenkins/plugins/analysis/warnings/CodeNArc.java
 src/main/java/io/jenkins/plugins/analysis/warnings/CppCheck.java
 src/main/java/io/jenkins/plugins/analysis/warnings/DocFx.java
 src/main/java/io/jenkins/plugins/analysis/warnings/ErrorProne.java
 src/main/java/io/jenkins/plugins/analysis/warnings/Flake8.java
 src/main/java/io/jenkins/plugins/analysis/warnings/JsHint.java
 src/main/java/io/jenkins/plugins/analysis/warnings/KlocWork.java
 src/main/java/io/jenkins/plugins/analysis/warnings/MyPy.java
 src/main/java/io/jenkins/plugins/analysis/warnings/Pit.java
 src/main/java/io/jenkins/plugins/analysis/warnings/PyDocStyle.java
 src/main/java/io/jenkins/plugins/analysis/warnings/ResharperInspectCode.java
 src/main/java/io/jenkins/plugins/analysis/warnings/Xlc.java
 src/main/java/io/jenkins/plugins/analysis/warnings/XmlLint.java
 src/main/java/io/jenkins/plugins/analysis/warnings/ZptLint.java
 src/main/resources/io/jenkins/plugins/analysis/warnings/Messages.properties
 src/test/java/hudson/plugins/warnings/parser/ResharperInspectCodeAdapterTest.java
 src/test/java/hudson/plugins/warnings/parser/ResharperInspectCodeParserTest.java
 src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java
 src/test/resources/io/jenkins/plugins/analysis/warnings/android-lint.xml
 src/test/resources/io/jenkins/plugins/analysis/warnings/codeNarc.xml
 src/test/resources/io/jenkins/plugins/analysis/warnings/cppcheck.xml
 src/test/resources/io/jenkins/plugins/analysis/warnings/docfx.json
 src/test/resources/io/jenkins/plugins/analysis/warnings/error-prone.log
 src/test/resources/io/jenkins/plugins/analysis/warnings/flake8.txt
 src/test/resources/io/jenkins/plugins/analysis/warnings/jshint.xml
 src/test/resources/io/jenkins/plugins/analysis/warnings/klocwork.xml
 src/test/resources/io/jenkins/plugins/analysis/warnings/mypy.txt
 src/test/resources/io/jenkins/plugins/analysis/warnings/pit.xml
 src/test/resources/io/jenkins/plugins/analysis/warnings/pydocstyle.txt
 src/test/resources/io/jenkins/plugins/analysis/warnings/spotbugsXml.xml
 src/test/resources/io/jenkins/plugins/analysis/warnings/xmllint.txt
 src/test/resources/io/jenkins/plugins/analysis/warnings/zptlint.log
<a href="http://jenkins-ci.org/commit/warnings-plugin/dcbd8e1b3b660ffa3764486557491849e77d63d9" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/warnings-plugin/dcbd8e1b3b660ffa3764486557491849e77d63d9</a>
Log:
  Let all parsers not depend on Jenkins anymore.</p>

<p>Integrated all parsers into new module analysis-model. Added adapter
classes for parsers of violations-lib.</p>
<ul class="alternate" type="square">
	<li><span class="error">&#91;FIXED JENKINS-17434&#93;</span></li>
	<li><span class="error">&#91;FIXED JENKINS-40439&#93;</span></li>
</ul>



<p>Compare: <a href="https://github.com/jenkinsci/warnings-plugin/compare/3f29bdbcfac9...dcbd8e1b3b66" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/warnings-plugin/compare/3f29bdbcfac9...dcbd8e1b3b66</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Ulli Hafner<br/>
Path:<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/AndroidLint.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/CodeNArc.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/CppCheck.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/DocFx.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/ErrorProne.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/Flake8.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/JsHint.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/KlocWork.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/MyPy.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/Pit.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/PyDocStyle.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/ResharperInspectCode.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/Xlc.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/XmlLint.java<br/>
 src/main/java/io/jenkins/plugins/analysis/warnings/ZptLint.java<br/>
 src/main/resources/io/jenkins/plugins/analysis/warnings/Messages.properties<br/>
 src/test/java/hudson/plugins/warnings/parser/ResharperInspectCodeAdapterTest.java<br/>
 src/test/java/hudson/plugins/warnings/parser/ResharperInspectCodeParserTest.java<br/>
 src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/android-lint.xml<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/codeNarc.xml<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/cppcheck.xml<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/docfx.json<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/error-prone.log<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/flake8.txt<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/jshint.xml<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/klocwork.xml<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/mypy.txt<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/pit.xml<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/pydocstyle.txt<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/spotbugsXml.xml<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/xmllint.txt<br/>
 src/test/resources/io/jenkins/plugins/analysis/warnings/zptlint.log<br/>
<a href="http://jenkins-ci.org/commit/warnings-plugin/dcbd8e1b3b660ffa3764486557491849e77d63d9" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/warnings-plugin/dcbd8e1b3b660ffa3764486557491849e77d63d9</a><br/>
Log:<br/>
  Let all parsers not depend on Jenkins anymore.</p>

<p>Integrated all parsers into new module analysis-model. Added adapter<br/>
classes for parsers of violations-lib.</p>
<ul class="alternate" type="square">
	<li><span class="error">&#91;FIXED <a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/17434">JENKINS-17434</a>&#93;</span></li>
	<li><span class="error">&#91;FIXED <a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/40439">JENKINS-40439</a>&#93;</span></li>
</ul>



<p>Compare: <a href="https://github.com/jenkinsci/warnings-plugin/compare/3f29bdbcfac9...dcbd8e1b3b66" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/warnings-plugin/compare/3f29bdbcfac9...dcbd8e1b3b66</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-40439?focusedId=349059&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349059">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Released in 5.0.0-beta2.</p>
</pre>
</details>

</ul>
</details>

<p>Released in 5.0.0-beta2.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a class="jira-relationship" href="https://issue-redirect.jenkins.io/issue/49911">JENKINS-49911</a>]</i>
